src/transform.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <glm/glm.hpp>
+
+/*
+    Handles all matrix transformations for a game object. Gives a (hopefully) simple
+    interface for interacting with the game object's transformation, including
+    translations, rotations, and scaling. 
+*/
+class Transform {
```

Comment:
For this class (and classes similar to it), we can consider using mixins. http://www.jakesee.com/2009/04/c-mixins-a-powerful-concept/